server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -131,6 +132,69 @@ private void update(MessageId messageId, UpdateMessagePatch updateMessagePatch,
 
     }
 
+    private void assertValidUpdate(List<MessageResult> messagesToBeUpdated, UpdateMessagePatch updateMessagePatch, MailboxSession session) throws MailboxException {
+        List<MailboxId> draftMailboxes = mailboxIdFor(Role.DRAFTS, session);
+        List<Flags> futureFlags = patchFlags(messagesToBeUpdated, updateMessagePatch);
+        List<MailboxId> targetMailboxes = getTargetedMailboxes(messagesToBeUpdated, updateMessagePatch);
+
+        boolean targetHasDraft = targetMailboxes.stream().anyMatch(draftMailboxes::contains);
+        boolean targetHasNonDraft = targetMailboxes.stream().anyMatch(id -> !draftMailboxes.contains(id));
+
+        assertValidUpdate(futureFlags, targetHasDraft, targetHasNonDraft);
+    }
+
+    private void assertValidUpdate(List<Flags> futureFlags, boolean targetHasDraft, boolean targetHasNonDraft) throws DraftMessageMailboxUpdateException {
+        assertNoMessageIntersetingDraftMailboxes(targetHasDraft, targetHasNonDraft);
```

Comment:
you simply mean assertDraftIsTheSingleTarget?

Reply:
I mean

`assertMessageNotDraftAndNonDraftMailboxes`

Reply:
are you sure?

Reply:
Yes, seems better to me ;-)

Reply:
Maybe more `assertMessageIsNotInDraftAndNonDraftMailboxes` to be more grammatically correct.

Reply:
ok, I don't know why you don't like my proposal, but ok for this one :)

Reply:
I don't like it because a set of non draft mailbox is valid, but it don't appear in your naming.